src/main/java/net/sacredlabyrinth/phaed/dynmap/simpleclans/managers/CommandManager.java

```diff
@@ -1,137 +1,153 @@
-		plugin.getConfig().getStringList("help-command").forEach(s -> sender.sendMessage(ChatColor.translateAlternateColorCodes('&', s)));
-	}
-
-	private boolean processToggleCommand(Player player) {
-		if (player.hasPermission("simpleclans.map.toggle")) {
-			PlayerEntry entry = plugin.getPlayerManager().getEntry(player);
-
-			if (entry.isVisible()) {
-				entry.setVisible(false);
-				player.sendMessage(plugin.getLang("not-visible"));
-			} else {
-				entry.setVisible(true);
-				player.sendMessage(plugin.getLang("visible"));
-			}
-		} else {
-			player.sendMessage(plugin.getLang("no-permission"));
-		}
-
-		return true;
-	}
-
-	private boolean processListIconsCommand(Player player) {
-		if (!player.hasPermission("simpleclans.map.list")) {
-			player.sendMessage(plugin.getLang("no-permission"));
-			return true;
-		}
-
-		player.sendMessage(plugin.getLang("available-icons"));
-		Set<String> icons = plugin.getClanHomes().getIcons();
-		icons.forEach(icon -> player.sendMessage(plugin.getLang("icon-line").replace("@icon", icon)));
-		if (icons.isEmpty()) {
-			player.sendMessage(plugin.getLang("error-no-icons"));
-		}
-
-		return true;
-	}
-
-	private boolean processReloadCommand(@NotNull CommandSender sender) {
-		if (!sender.hasPermission("simpleclans.map.reload")) {
-			sender.sendMessage(plugin.getLang("no-permission"));
-			return true;
-		}
-
-		sender.sendMessage(plugin.getLang("reloading"));
-		plugin.cleanup();
-		PreferencesManager.loadPreferences();
-		plugin.reloadConfig();
-		plugin.activate();
-
-		return true;
-	}
-
-	private boolean processSetIconCommand(@NotNull Player player, @NotNull String icon) {
-		ClanPlayer cp = plugin.getClanManager().getClanPlayer(player);
-		if (cp == null || cp.getClan() == null) {
-			player.sendMessage(plugin.getLang("not-member"));
-			return true;
-		}
-		if (!player.hasPermission("simpleclans.map.seticon") || !cp.isLeader()) {
-			player.sendMessage(plugin.getLang("no-permission"));
-			return true;
-		}
-
-		if (plugin.getClanHomes().getIcons().contains(icon.toLowerCase())) {
-			if (!player.hasPermission("simpleclans.map.icon.bypass") &&
-					!player.hasPermission("simpleclans.map.icon." + icon)) {
-				player.sendMessage(plugin.getLang("no-permission"));
-				return true;
-			}
-			PreferencesManager pm = new PreferencesManager(cp.getClan());
-			pm.setClanHomeIcon(icon);
-			player.sendMessage(plugin.getLang("icon-changed"));
-		} else {
-			player.sendMessage(plugin.getLang("icon-not-found"));
-		}
-
-		return true;
-	}
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static net.sacredlabyrinth.phaed.dynmap.simpleclans.DynmapSimpleClans.lang;
+
+public final class CommandManager implements TabExecutor {
+
+    private final @NotNull DynmapSimpleClans plugin;
+    private final @Nullable HomeLayer homeLayer;
```

Comment:
If the plugin is reloaded, this will not be updated

Reply:
[This](https://www.spigotmc.org/threads/how-to-register-events-on-server-reload.386916/#post-3494059) guy told it will call onEnable() after reload

Reply:
`plugin.reload()` will not call onEnable().
I mean if DSC's reload command is used, not the server's.

Reply:
> I mean if DSC's reload command is used, not the server's.
It will be updated.
Since it's called `plugin.reload` on DSC`s reload command, it will run `loadLayers()`, where they will be instantiated again.

Reply:
loadLayers() doesn't update the value of the field in CommandManager.